src/interfaces/chain.h

```diff
@@ -271,8 +260,10 @@ class Chain
     //! Register handler for notifications.
     virtual std::unique_ptr<Handler> handleNotifications(Notifications& notifications) = 0;
 
-    //! Wait for pending notifications to be handled.
-    virtual void waitForNotifications() = 0;
+    //! Wait for pending notifications to be processed unless block hash points to the current,
+    //! chain tip, or to a possible descendant of the current chain tip that isn't currently
+    //! connected.
+    virtual void waitForNotificationsUpToTip(const uint256& hash) = 0;
```

Comment:
I think the "wait up to" name I suggested earlier in the todo comment is misleading, because this isn't really deciding what to wait for based on the argument, but deciding whether to wait based on the argument.

I think a better name might be `waitForNotificationsIfNewBlocks(old_tip)` or `waitForNotificationsIfNewBlocksConnected(old_tip)`

Reply:
Yes agree wasn't clear so I added `Tip` but even followed `waitForNotificationsIfNewBlocksConnected(old_tip)` suggestion, a little bit more verbose but far more insightful